pkgs/by-name/ro/root/package.nix

```diff
@@ -81,37 +81,37 @@ stdenv.mkDerivation rec {
     pkg-config
     git
   ];
-  propagatedBuildInputs = [ nlohmann_json ];
```

Comment:
I think this one is a regression:
```
CMake Error at /nix/store/pcbl527aznd07y7109mbhkv1za05bfhj-root-6.34.00/share/root/cmake/ROOTConfig-targets.cmake:763 (set_target_properties):
  The link interface of target "ROOT::ROOTEve" contains:

    nlohmann_json::nlohmann_json

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /nix/store/pcbl527aznd07y7109mbhkv1za05bfhj-root-6.34.00/share/root/cmake/ROOTConfig.cmake:113 (include)
  CMakeLists.txt:127 (find_package)
```

Reply:
https://github.com/NixOS/nixpkgs/pull/366357

Reply:
How did you get this error? The thing is that `nlohmann_json` is only a dependency when you're explicitly requiring the `REve` component. See also: https://github.com/root-project/root/pull/16703 You're using that? If not, but you still get the error, it's a bug in ROOT.

If you *are* using REve but you get this error, then can this be fixed by specifying `nlohmann_json` as an input for your project?

In other words, `nlohmann_json` is only an optional propagated dependency when one uses REve, which is not very common. Therefore I was thinking it is not always worth to propagate this input.

Reply:
Yes, this is REve being used in DD4hep. Looks like they got burnt by the change in behaviour: https://github.com/AIDASoft/DD4hep/commit/bf69d07c6414a4a0d03bae02546ac2d2dcd7f30b

For a typical nixpkgs package, I would expect that, if some feature exposed by the package, the transient dependencies would be provided by default.

Also want to check with you. Looking at https://github.com/root-project/root/blob/a4a2590c97a1f31f254bfa112f8f4f5326264d51/graf3d/eve7/CMakeLists.txt#L144-L148, it looks like with `builtin_nlohmannjson=ON` the include path is either privated or propagated elsewhere implicitly. Is it really part of Eve's API?

Reply:
Ok, then let's propagate the `nlohmann_json` input in any case, like you suggested in your PR:

Yes, `nlohmann_json` is part of the public REve API:
https://github.com/root-project/root/blob/a4a2590c97a1f31f254bfa112f8f4f5326264d51/graf3d/eve7/inc/ROOT/REveElement.hxx#L26

It's to support JSON IO of custom REve elements. Maybe there would have been a way to abstact the JSON library away better, but this is how it is for now so we have to deal with it